okhttp/src/commonJvmAndroid/kotlin/okhttp3/internal/http/RealInterceptorChain.kt

```diff
@@ -135,4 +161,39 @@ class RealInterceptorChain(
 
     return response
   }
+
+  /**
+   * Creates an [Address] of out of the provided [HttpUrl]
+   * that uses this client’s DNS, TLS, and proxy configuration.
+   */
+  fun address(url: HttpUrl): Address {
+    var useSslSocketFactory: SSLSocketFactory? = null
+    var useHostnameVerifier: HostnameVerifier? = null
+    var useCertificatePinner: CertificatePinner? = null
+    if (url.isHttps) {
+      useSslSocketFactory = call.client.sslSocketFactory
+      useHostnameVerifier = call.client.hostnameVerifier
+      useCertificatePinner = call.client.certificatePinner
+    }
+
+    return Address(
+      uriHost = url.host,
+      uriPort = url.port,
+      dns = clientOverrides?.dns ?: call.client.dns,
+      socketFactory = clientOverrides?.socketFactory ?: call.client.socketFactory,
+      sslSocketFactory = useSslSocketFactory,
+      hostnameVerifier = useHostnameVerifier,
+      certificatePinner = useCertificatePinner,
+      proxyAuthenticator = call.client.proxyAuthenticator,
+      proxy = call.client.proxy,
+      protocols = call.client.protocols,
+      connectionSpecs = call.client.connectionSpecs,
+      proxySelector = call.client.proxySelector,
+    )
+  }
+
+  data class ClientOverrides(
```

Comment:
I think I’d prefer adding fields to RealInterceptorChain. It avoids having to build a complex mechanism to support chaining ClientOverrides